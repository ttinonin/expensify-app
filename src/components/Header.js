import React from "react";
import { NavLink } from 'react-router-dom';

const Header = () => (
    <header className="header">
        <h1>Expensify!</h1>
        <ul>
            <NavLink to='/' activeClassName="is-active" exact={true}><li>Home</li></NavLink>
            <NavLink to='/create' activeClassName="is-active"><li>Create</li></NavLink>
            <NavLink to='/edit' activeClassName="is-active"><li>Edit</li></NavLink>
        </ul>
    </header>
)

export default Header