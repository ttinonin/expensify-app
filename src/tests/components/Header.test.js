import { shallow } from 'enzyme'
import React from 'react'
import Header from '../../components/Header'

//tira uma "print" do codigo do header
//cada vez que eu mudar ele o jest me avisa

test('should render header correctly',  ()=>{
    const wrapper = shallow(<Header />)
    expect(wrapper.find('h1').length).toBe(1)
    
    //const renderer = new ReactShallowRenderer()
    //renderer.render(<Header />)
    //expect(renderer.getRenderOutput()).toMatchSnapshot()
})