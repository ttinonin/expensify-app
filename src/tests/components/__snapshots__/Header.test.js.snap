// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<header
  className="header"
>
  <h1>
    Expensify!
  </h1>
  <ul>
    <NavLink
      activeClassName="is-active"
      exact={true}
      to="/"
    >
      <li>
        Home
      </li>
    </NavLink>
    <NavLink
      activeClassName="is-active"
      to="/create"
    >
      <li>
        Create
      </li>
    </NavLink>
    <NavLink
      activeClassName="is-active"
      to="/edit"
    >
      <li>
        Edit
      </li>
    </NavLink>
  </ul>
</header>
`;
